api-reference/10 UI Components/dxDataGrid/1 Configuration/summary/calculateCustomSummary.md

```diff
@@ -113,13 +118,353 @@ A summary value calculation is conducted in three stages: *start* - the **totalV
+                            break;
+                        case "calculate":
+                            // Modifying "totalValue" here
+                            break;
+                        case "finalize":
+                            // Assigning the final value to "totalValue" here
+                            break;
+                    }
+                }
+
+                // Calculating "customSummary2"
+                if(options.name == "customSummary2") {
+                    // ...
+                    // Same "switch" statement here
+                }
+            },
+        },
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import DataGrid, { Column, TotalItem } from 'devextreme-react/data-grid';
+
+    const calculateCustomSummary = (options) => {
+        // Calculating "customSummary1"
+        if(options.name == "customSummary1") {
+            switch(options.summaryProcess) {
+                case "start":
+                    // Initializing "totalValue" here
+                    break;
+                case "calculate":
+                    // Modifying "totalValue" here
+                    break;
+                case "finalize":
+                    // Assigning the final value to "totalValue" here
+                    break;
+            }
+        }
+
+        // Calculating "customSummary2"
+        if(options.name == "customSummary2") {
+            // ...
+            // Same "switch" statement here
+        }
+    }
+
+    function App() {
+        return (
+            <DataGrid ... >
+                <Summary calculateCustomSummary={calculateCustomSummary}>
+                    <TotalItem
+                        name="customSummary1"
+                        summaryType="custom"
+                    />
+                    <TotalItem
+                        name="customSummary2"
+                        summaryType="custom"
+                    />
+                </Summary>
+            </DataGrid>
+        );
+    }
+
+    export default App;
+
+---
+
+You can use the **value** field to access a column value. If you do not specify a [column](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/summary/totalItems/#column) by which to calculate the summary, the **value** field contains an entire data object. However, this object misses values from unbound columns calculated in the [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue) function. If you need these values for your custom summary, call the **calculateCellValue** function from inside **calculateCustomSummary**, as shown below. In this example, the calculateArea (calculateCellValue) function creates an unbound column 'Area'. The same function is called from the calculateAreaSummary (calculateCustomSummary) function to compute the sum of areas for selected rows.
```

Comment:
- What do you mean when you say "to access"? To get the value? To retrieve the value?
- "by which to calculate the summary" Do you mean 'used to calculate the summary' or 'for which to calculate the summary'?